src/cli.rs

```diff
@@ -1,14 +1,195 @@
+        let _ = fs::remove_dir_all(&local_binary_project_src_directory);
+        let _ = fs::remove_file(&local_binary_project_cargo_toml_path);
+        let _ = fs::remove_file(local_binary_project_directory.join("Cargo.lock"));
+    }
+    fs::create_dir_all(local_binary_project_directory)
+        .expect("Couldn't create local binary project directory");
+    let has_local_rules = local_binary_project_directory
+        .parent()
+        .unwrap()
+        .join(LOCAL_RULES_DIR_NAME)
+        .is_dir();
+
+    let cargo_toml_contents = get_local_binary_cargo_toml_contents(
+        &parsed_config_file,
+        has_local_rules,
+        relative_path_from_local_binary_project_directory_to_project_directory,
+    );
+    fs::write(local_binary_project_cargo_toml_path, cargo_toml_contents)
+        .expect("Couldn't write local binary project Cargo.toml");
+
+    fs::create_dir(&local_binary_project_src_directory)
+        .expect("Couldn't create local binary project `src/` directory");
+    let src_main_rs_contents = get_src_main_rs_contents(&parsed_config_file, has_local_rules);
+    fs::write(
+        local_binary_project_src_directory.join("main.rs"),
+        src_main_rs_contents,
+    )
+    .expect("Couldn't write local binary project src/main.rs");
+
+    let gitignore_contents = get_gitignore_contents();
+    fs::write(
+        local_binary_project_directory.join(".gitignore"),
+        gitignore_contents,
+    )
+    .expect("Couldn't write local binary project .gitignore");
+
+    release_build_local_binary(local_binary_project_directory);
+}
+
+fn release_build_local_binary(local_binary_project_directory: &Path) {
+    let output = Command::new("cargo")
+        .args(["build", "--release"])
+        .current_dir(local_binary_project_directory)
+        .output()
+        .expect("Failed to execute cargo release build command");
+    if !output.status.success() {
+        panic!("Cargo release build of local binary project failed");
+    }
+}
+
+fn get_local_binary_cargo_toml_contents(
+    parsed_config_file: &ParsedConfigFile,
+    has_local_rules: bool,
+    relative_path_from_local_binary_project_directory_to_project_directory: &Path,
+) -> String {
+    let mut contents = String::new();
+    contents.push_str("[package]\n");
+    contents.push_str(&format!("name = \"{}\"\n", LOCAL_BINARY_PROJECT_NAME));
+    contents.push_str("version = \"0.1.0\"\n");
+    contents.push_str("edition = \"2021\"\n\n");
+    contents.push_str("[dependencies]\n");
+    contents.push_str("tree-sitter-lint = { path = \"../..\" }\n");
+    if has_local_rules {
+        contents.push_str(&format!(
+            "local_rules = {{ path = \"../{}\" }}\n",
+            LOCAL_RULES_DIR_NAME
+        ));
+    }
+    for (plugin, plugin_spec) in &parsed_config_file.content.plugins {
+        let path = plugin_spec
+            .path
+            .as_ref()
+            .expect("Currently only handling local path plugin dependencies");
```

Comment:
Generating the `Cargo.toml` dependencies for plugins that are on crates.io is interesting as far as what version to put?

I guess I was picturing "avoiding having to figure that out" by just having it run `cargo add` for any such dependencies if there's not a straight-forward way to figure out what the version should be ourselves?